### PR TITLE
[doc, aon_timer] Clarify when wakeup and interrupt triggered

### PR DESCRIPTION
**Pull request by GregAC**
Explicitly state wakeup and watchdog counts must be cleared along with
the wakeup and interrupt events to avoid immediate retriggering.

Signed-off-by: Greg Chadwick <gac@lowrisc.org>

### CODE REVIEW

**Comment by GregAC** [comment]
This is a doc fix only so we can get it merged to enable aon_timer v1. RTL fix requires some chip-level test fixes which may take time to review: https://github.com/lowRISC/opentitan/pull/10221

File: hw/ip/aon_timer/doc/_index.md [hunk 1]
```diff
@@ -20,8 +20,10 @@ See that document for an overview of how it is integrated into the top level sys
 
 The always-on wakeup timer operation is straightforward.
 A count starts at 0 and slowly ticks upwards (one tick every `N + 1` clock cycles, where `N` is the pre-scaler value).
-When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is sent to the processor.
-This wakeup signal stays high until it is explicitly acknowledged by software (software must write zero to the {{<regref "WKUP_CAUSE">}} register to clear it).
+When it reaches / exceeds the wake threshold, a level wakeup signal is sent to the power manager and a level IRQ is sent to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+Software must write 0 to the {{<regref "WKUP_CAUSE">}} register to clear the wakeup and 1 to {{<regref "INTR_STATE.wkup_timer_expired">}} to clear the interrupt.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:25]: Nit: Maybe "... and then 1 to ..." to make it clear that it has to happen in this order (to stop the interrupt re-triggering)?

File: hw/ip/aon_timer/doc/_index.md [hunk 2]
```diff
@@ -20,8 +20,10 @@ See that document for an overview of how it is integrated into the top level sys
...
+Note that if {{<regref "WKUP_COUNT">}} is not zeroed and remains at or above the wake threshold or the wakeup timer isn't disabled the wakeup and interrupt will trigger again at the next clock tick.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:26]: I'd suggest a comma here: ".. or the wakeup timer isn't disabled, the wakeup and ...". Otherwise it's a bit difficult to parse the sentence.

File: hw/ip/aon_timer/doc/_index.md [hunk 3]
```diff
@@ -103,6 +105,8 @@ Clear the interrupt by writing 1 into the Interrupt Status Register {{<regref "I
 
 If the timer has caused a wakeup event ({{<regref"WKUP_CAUSE">}} is set) then clear the wakeup request by writing 0 to {{<regref"WKUP_CAUSE">}}.
 
+If {{<regref"WKUP_COUNT">}} remains above the threshold after clearing the interrupt or the wakeup event and it remains enabled it will trigger again at the next clock tick.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:108]: Again, this is pretty hard to parse and I'm not sure what "it" is for the thing that remains enabled. Maybe this?

```
If {{<regref"WKUP_COUNT">}} remains above the threshold after clearing the interrupt or wakeup event and the timer remains enabled, the interrupt will trigger again at the next clock tick.
```

**Review by rswarbrick — APPROVED** [review]
Grammar nits, but the changes look good to me.

File: hw/ip/aon_timer/doc/_index.md (hunk 1 above)
Comment (GregAC) [reply hw/ip/aon_timer/doc/_index.md:25]: They're actually not related. The wakeup signal is going to the power manager, the interrupt is going to the Ibex external IRQ. I'll rewrite into two sentences to make it clear they're separate.

File: hw/ip/aon_timer/doc/_index.md (hunk 3 above)
Comment (GregAC) [reply hw/ip/aon_timer/doc/_index.md:108]: Sounds good, made a slight adjustment as interrupt and wakeup are two seperate things (both of which will get retriggered if you don't do at least one of clear count or disable timer).

File: hw/ip/aon_timer/doc/_index.md (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/aon_timer/doc/_index.md]: Oh! That makes sense, thanks.

File: hw/ip/aon_timer/doc/_index.md [hunk 4]
```diff
@@ -20,8 +20,11 @@ See that document for an overview of how it is integrated into the top level sys
 
 The always-on wakeup timer operation is straightforward.
 A count starts at 0 and slowly ticks upwards (one tick every `N + 1` clock cycles, where `N` is the pre-scaler value).
-When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is sent to the processor.
-This wakeup signal stays high until it is explicitly acknowledged by software (software must write zero to the {{<regref "WKUP_CAUSE">}} register to clear it).
+When it reaches / exceeds the wake threshold, a level wakeup signal is sent to the power manager and a level IRQ is sent to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+To clear the wakeup write 0 to the {{<regref "WKUP_CAUSE">}} register.
+To clear the interrupt write 1 to {{<regref "INTR_STATE.wkup_timer_expired">}}.
+Note that if {{<regref "WKUP_COUNT">}} is not zeroed and remains at or above the wake threshold or the wakeup timer isn't disabled, the wakeup and interrupt will trigger again at the next clock tick.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:27]: Should this be "**and** the wakeup timer"?

**Review by rswarbrick — APPROVED** [review]
Looks good to me, with one question.
